core/include/drivers/wdt.h

```diff
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2019 Broadcom.
+ */
+
+#ifndef DRIVERS_WDT_H
+#define DRIVERS_WDT_H
+
+#include <kernel/interrupt.h>
+#include <tee_api_types.h>
+
+struct wdt_chip {
+	const struct wdt_ops *ops;
+	struct itr_handler wdt_itr;
```

Comment:
As interrupt support is optional, maybe this field could a memory ref, rather than the structure itself.
Could be allocated from `sp805_register_itr_handler()`, or refer to caller structure.

Reply:
ok, will make memory ref.